File: app/src/main/java/com/maciega/bartosz/crawl/model/SpiderServantActions.java
package com.maciega.bartosz.crawl.model;

import java.util.List;

import io.realm.RealmObject;

/**
 * Created by Bartosz on 2016-10-18.
 */

public interface SpiderServantActions {
    void saveUrl(Url url);
    void saveUrls(List<Url> list);
    void deleteUrl(Url url);
    void deleteAll(Url url);
    void clearDb();
    void  getUrls();
    void getUrl(String primaryKey);
}

File: app/src/main/java/com/maciega/bartosz/crawl/model/SpiderServantActions.kt
package com.maciega.bartosz.crawl.model

/**
 * Created by Bartosz on 2016-10-18.
 */

interface SpiderServantActions {
    fun saveUrl(url: Url)
    fun saveUrls(list: List<Url>)
    fun deleteUrl(url: Url)
    fun deleteAll(url: Url)
    fun clearDb()
    fun getUrls()
    fun getUrl(primaryKey: String)
}
